default.nix

```diff
@@ -135,20 +135,16 @@ let
 
     # Produce a fixed output derivation from a moving target (hackage index tarball)
     hackageTarball = { index-state, sha256 }:
-      pkgs.runCommand "01-index.tar.gz-at-${builtins.replaceStrings [":"] [""] index-state}" {
-        nativeBuildInputs = [ pkgs.curl ];
+      assert sha256 != null;
+      pkgs.fetchurl {
+        name = "01-index.tar.gz-at-${builtins.replaceStrings [":"] [""] index-state}";
```

Comment:
maybe drop the `.tar.gz` and add `recursiveHash = true;`
also, is `$out` a file or directory?, its not clear which `truncate-index` is creating

Reply:
@cleverca22 `truncate-index` takes a `.tar.gz` and produces a `.tar.gz` So it should be a single file.